src-self-hosted/translate_c.zig

```diff
@@ -310,6 +302,59 @@ fn visitFnDecl(c: *Context, fn_decl: *const ZigClangFunctionDecl) Error!void {
     return addTopLevelDecl(c, fn_name, &proto_node.base);
 }
 
+fn visitVarDecl(c: *Context, var_decl: *const ZigClangVarDecl) Error!void {
+    if (try c.decl_table.put(@ptrToInt(var_decl), {})) |_| return; // Avoid processing this decl twice
+    const rp = makeRestorePoint(c);
+    var scope = &c.global_scope.base;
+    const var_name = try c.str(ZigClangDecl_getName_bytes_begin(@ptrCast(*const ZigClangDecl, var_decl)));
+    const var_decl_loc = ZigClangVarDecl_getLocation(var_decl);
+
+    switch (ZigClangVarDecl_getTLSKind(var_decl)) {
+        .None => {},
+        .Static => return failDecl(c, var_decl_loc, var_name, "static thread local storage not supported", .{}),
+        .Dynamic => return failDecl(c, var_decl_loc, var_name, "dynamic thread local storag not supported", .{}),
+    }
+
+    const qual_type = ZigClangVarDecl_getType(var_decl);
+    const is_extern = ZigClangVarDecl_hasExternalStorage(var_decl);
+    const is_static = ZigClangVarDecl_isFileVarDecl(var_decl);
+    const is_const = ZigClangQualType_isConstQualified(qual_type);
+
+    var var_node = try transCreateNodeVarDecl(c, true, is_extern, is_const, var_name);
+
+    _ = try appendToken(rp.c, .Colon, ":");
+    var_node.type_node = transQualType(rp, qual_type, var_decl_loc) catch |err| switch (err) {
+        error.UnsupportedType => {
+            return failDecl(c, var_decl_loc, var_name, "unable to resolve variable type", .{});
+        },
+        error.OutOfMemory => |e| return e,
+    };
+
+    if (is_static and !is_extern) {
+        const eq_tok = try appendToken(c, .Equal, "=");
+        const init_node = if (ZigClangVarDecl_hasInit(var_decl)) blk: {
+            const ap_value = ZigClangVarDecl_evaluateValue(var_decl) orelse
+                return failDecl(c, var_decl_loc, var_name, "unable to evaluate initializer", .{});
+            break :blk transApValue(rp, ap_value, qual_type, var_decl_loc) catch |err| switch (err) {
```

Comment:
This should probably just use `ZigClangVarDecl_getInit` and then use`transExpr`.